packages/common/src/index.ts

```diff
@@ -780,9 +780,9 @@ export default class Common extends EventEmitter {
     // Logic: if accumulator is still null and on the first occurrence of
     // a block greater than the current hfBlock set the accumulator,
     // pass on the accumulator as the final result from this time on
-    const nextHfBlock = this.hardforks().reduce((acc: BN | null, hf: HardforkParams) => {
-      const block = new BN(hf.block!)
-      return block.gt(hfBlock) && acc === null ? block : acc
+    const nextHfBlock = this.hardforks().reduce((acc: bigint | null, hf: HardforkParams) => {
+      const block = BigInt(hf.block === null ? 0 : hf.block)
+      return block > hfBlock && acc === null ? block : acc
```

Comment:
I have a bit my problems to read this code TBH since I am not so familiar with the `reduce()` functionality. Just to mention here: so the return value for a HF where block is `null` (this means: HF in the future, e.g. Merge/Shanghai) should not be BigInt(0) but `null`. Is this the case here (should be covered by tests though)? 🤔